support-frontend/assets/helpers/customHooks/analyticsProfileCache.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * In-memory cache for analytics profile data using React Context.
+ * This cache is shared across all hook instances within the same provider tree,
+ * preventing duplicate API calls without requiring cookies or localStorage.
+ * Data is cleared when the page is refreshed (no GDPR concerns).
+ */
+
+import { createContext, type ReactNode, useContext, useRef } from 'react';
+
+interface AnalyticsProfileData {
+	hasMobileAppDownloaded: boolean;
+	hasFeastMobileAppDownloaded: boolean;
+}
+
+interface CacheEntry {
+	data: AnalyticsProfileData;
+	timestamp: number;
+}
+
+class AnalyticsProfileCache {
```

Comment:
I am not completely up to speed on what you can and can't do on the client side, and where Promises are allowed, but why do we need both get and getPendingRequest? 
Can we have one method get, which returns a Promise<AnalyticsProfileData> and will either return a new one if expired, or a stored one otherwise?

Reply:
oh, it looks like it created some context for sharing a singleton class instance instead of just using some context to store the data and possibly if it is in a pending state.

Maybe we could revert to just using the singleton class on it's own for now and then look into using only react context and no singleton class later on as an improvement?